EIPS/eip-2390.md

```diff
@@ -0,0 +1,297 @@
+from MaxMind which maps the IP address to a location.
+This method of locating the source of a query is error prone and unreliable.
+If the GeoIP database is out of date, queried locations can be vastly different than their true location.
+GeoENS does not rely on a database because the user includes a location in their query.
+
+It follows that queries can be made by users for any location, not just their location.
+Traditional DNS will only return the resource assigned to a query's provenance.
+GeoENS does not correlate a query's provinance with a location, allowing the
+entire globe to be queried from a single location.
+
+An additional shortcoming of traditional DNS is the fact that there is no way to return a list of servers in a certain proximity.
+This is paramount for uses cases that require discovering the resource with the lowest latency.
+GeoENS allows a list of resources, like IP addresses, to be gathered within a specific location.
+Then a client to determine themselves which resource has the lowest latency.
+
+Lastly, publicly facing GeoDNS services do not give fine granularity control
+over geographic regions for GeoDNS queries.
+Cloud based DNS services like [Amazon's Route 53](https://aws.amazon.com/route53/)
+only allow specifying geographic regions at the granularity of a State in
+the United States.
+GeoENS on the other hand gives 8 characters of geohash resolution which
+corresponds to +-20 meter accuracy.
+
+
+## Specification
+This EIP proposes a new interface to ENS resolvers such that geo-spacial information
+can be recorded and retrieved from the blockchain.
+The interface changes are described below for "address resolvers" described in EIP137
+however the idea applies to any record described in EIP1185 and EIP1062, namely DNS
+Resolvers, Text Resolvers, ABI Resolvers, etc.
+
+### What is a geohash?
+Geohashes are typically encoded in base 32 characters, as described
+https://en.m.wikipedia.org/wiki/Geohash#Algorithm_and_example.
+A geohash is an interleaving of latitude and longitude bits, whose
+length determines it's precision.
+
+
+### function setGeoAddr(bytes32 node, string calldata geohash, address addr) external authorised(node)
+Sets a resource (contract address, IP, ABI, TEXT, etc.) by node and geohash.
+Geohashes must be unique per address and are exactly 8 characters long.
+This leads to an accuracy of +-20 meters.
+Write default initialized resource value, `address(0)`, to remove a resource from the resolver.
+
+### function geoAddr(bytes32 node, string calldata geohash) external view returns (address[] memory ret)
+Query the resolver contract for a specific node and location.
+All resources (contract addresses, IP addresses, ABIs, TEXT records, etc.) matching
+the node and prefix geohash provided are returned.
+This permits querying by exact geohash of 8 characters to return the contect at that location,
```

Comment:
I don't think `contect` is a word?  Spellchecker doesn't think so at least...

Reply:
Fixed in 1b4d2c746a4653132a954160e43ea1e5e616326e